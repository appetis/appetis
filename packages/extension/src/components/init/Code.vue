<template>
  <div
    class="fixed inset-0 overflow-y-auto z-10"
    aria-labelledby="modal-title"
    role="dialog"
    aria-modal="true"
  >
    <div class="init-modal-background" aria-hidden="true"></div>
    <div class="modal">
      <!--      <span class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true">&#8203;</span>-->

      <div class="modal-content-xl">
        <div class="flex">
          <Left />
          <div class="w-6/12 relative">
            <Header title="Unlock with Invite Code" />
              <div class="modal-body">
                <p>Please enter the 6-digit code your team sent to your email to join your team dashboard in workly.</p>

                <form @submit.prevent="onSubmit">
                  <div class="flex mt-6">
                    <div class="w-1/6 px-1">
                      <input
                        type="text"
                        maxlength="1"
                        class="input-code"
                        ref="code_1"
                        @keyup="codeKeyup(1)"
                        @focus="onFocus(1)"
                      />
                    </div>
                    <div class="w-1/6 px-1">
                      <input
                        type="text"
                        maxlength="1"
                        class="input-code"
                        ref="code_2"
                        @keyup="codeKeyup(2)"
                        @focus="onFocus(2)"
                      />
                    </div>
                    <div class="w-1/6 px-1">
                      <input
                        type="text"
                        maxlength="1"
                        class="input-code"
                        ref="code_3"
                        @keyup="codeKeyup(3)"
                        @focus="onFocus(3)"
                      />
                    </div>
                    <div class="w-1/6 px-1">
                      <input
                        type="text"
                        maxlength="1"
                        class="input-code"
                        ref="code_4"
                        @keyup="codeKeyup(4)"
                        @focus="onFocus(4)"
                      />
                    </div>
                    <div class="w-1/6 px-1">
                      <input
                        type="text"
                        maxlength="1"
                        class="input-code"
                        ref="code_5"
                        @keyup="codeKeyup(5)"
                        @focus="onFocus(5)"
                      />
                    </div>
                    <div class="w-1/6 px-1">
                      <input
                        type="text"
                        maxlength="1"
                        class="input-code"
                        ref="code_6"
                        @keyup="codeKeyup(6)"
                        @focus="onFocus(6)"
                      />
                    </div>
                  </div>

                  <div class="mt-7">
                    <button class="btn-black-full" id="btn-code-continue">
                      Continue
                    </button>
                  </div>
                </form>
              </div>
              <div class="modal-footer init-modal-footer">
                Don't have an invite code?
                <span
                  class="text-blue cursor-pointer font-semibold underline"
                  @click="$emit('goSignup')"
                  id="text-register"
                  >Register</span
                >
                |
                <span
                  class="text-blue cursor-pointer font-semibold underline"
                  @click="$emit('goSignin')"
                  id="text-login"
                  >Login</span
                >
              </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Left from '@/components/init/Left'
import Header from '@/components/init/Header'

export default {
  name: 'Code',
  props: ['callFocus'],
  components: {
    Left,
    Header
  },
  data() {
    return {
      current: 1,
    }
  },
  mounted() {
    this.focusInput()
    // if(this.showCode) this.focusInput()
  },
  watch: {
    callFocus() {
      console.log('====>', 'callFocus')
      this.focusInput()
    },
  },
  methods: {
    focusInput() {
      //this.$refs['code_1'].focus();
      this.goNextFocus('code_1')
    },
    goNextFocus(next) {
      //console.log('===>', next)
      this.$refs[next].focus()
    },
    onFocus(num) {
      for (let i = 0; i < num; i++) {
        if (!this.$el.getElementsByClassName('input-code')[i].value) {
          this.current = i + 1
          break
        } else {
          if (i === num - 1) this.current = num
        }
      }
      this.goNextFocus('code_' + this.current)
    },
    setCode() {
      let code = ''
      for (let el_code of this.$el.getElementsByClassName('input-code')) {
        code += el_code.value.toUpperCase()
      }
      return code
    },
    codeKeyup(num) {
      this.current = num
      if (this.$refs['code_' + num].value) {
        if (this.current < 6) this.current++
      }
      // else {
      //   //if (this.current > 1) this.current--
      // }

      this.goNextFocus('code_' + this.current)
    },
    onSubmit() {
      const code = this.setCode()
      if (code.length < 6) return false
      //console.log(code)

      this.$store
        .dispatch('verifyCode', code)
        .then(() => {
          this.freshUserObject()
          console.log('success')
        })
        .catch((error) => {
          console.log(error)
        })
    },
  },
}
</script>
