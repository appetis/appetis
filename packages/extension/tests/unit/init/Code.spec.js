import Code from '@/components/init/Code'
import { mount } from '@vue/test-utils'

describe('init/Code', () => {
  it('code modal snapshot', () => {
    const wrapper = mount(Code)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('click continue button', async () => {
    const wrapper = mount(Code)
    const mockOnSubmit = jest.spyOn(wrapper.vm, 'onSubmit')
    await wrapper.find('form').trigger('submit')
    //await wrapper.find('#btn-code-continue').trigger('click')

    expect(mockOnSubmit).toHaveBeenCalled()
  })

  it('click Register text', () => {
    const wrapper = mount(Code)
    const mockGoSignup = jest.fn()
    wrapper.vm.$on('goSignup', mockGoSignup)
    //const mockGoSignup = jest.spyOn(wrapper.vm, 'goSignup')
    wrapper.find('#text-register').trigger('click')

    expect(mockGoSignup).toHaveBeenCalledTimes(1)
  })

  it('click Login text', () => {
    const wrapper = mount(Code)
    const mockGoSignin = jest.fn()
    wrapper.vm.$on('goSignin', mockGoSignin)
    //const mockGoSigin = jest.spyOn(wrapper.vm, 'goSignup')
    wrapper.find('#text-login').trigger('click')

    expect(mockGoSignin).toHaveBeenCalledTimes(1)
  })
})
