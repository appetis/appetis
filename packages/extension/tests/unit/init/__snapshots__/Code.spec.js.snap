// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init/Code code modal snapshot 1`] = `
<div aria-labelledby="modal-title" role="dialog" aria-modal="true" class="fixed inset-0 overflow-y-auto z-10">
  <div aria-hidden="true" class="init-modal-background"></div>
  <div class="modal">
    <div class="modal-content-xl">
      <div class="flex">
        <div class="w-7/12 bg-wk-dark p-14 text-white"><img src="" alt="workly logo" class="h-10">
          <p class="mt-3">
            We are all about the team. Organize projects, meetings, and individual schedules to get things done in time.
          </p>
          <div class="text-center mt-14"><img src="" alt="info"></div>
        </div>
        <div class="w-6/12 relative">
          <div class="modal-header"><img src="" alt="worlky small logo">
            <div class="modal-title">Unlock with Invite Code</div>
          </div>
          <div class="modal-body">
            <p>Please enter the 5-digit code your team sent to your email to join your team dashboard in workly.</p>
            <form>
              <div class="flex mt-6">
                <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
                <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
                <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
                <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
                <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
                <div class="w-1/6 px-1"><input type="text" maxlength="1" class="input-code"></div>
              </div>
              <div class="mt-7"><button id="btn-code-continue" class="btn-black-full">
                  Continue
                </button></div>
            </form>
          </div>
          <div class="modal-footer init-modal-footer">
            Don't have an invite code?
            <span id="text-register" class="text-blue cursor-pointer font-semibold underline">Register</span>
            |
            <span id="text-login" class="text-blue cursor-pointer font-semibold underline">Login</span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
